examples/seq2seq/seq2seq_trainer.py

```diff
@@ -122,7 +122,8 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
         else:
             if self.args.sortish_sampler:
                 self.train_dataset.make_sortish_sampler(
-                    self.args.per_device_train_batch_size, distributed=self.args.n_gpu > 1
+                    self.args.per_device_train_batch_size,
+                    distributed=(self.args.n_gpu > 1 and self.args.local_rank != -1),
```

Comment:
This is always `False` because of [this line](https://github.com/huggingface/transformers/blob/19fa01ce2acdffab84014fa99df2f107b67059ce/src/transformers/training_args.py#L431). The `n_gpu` property is (as documented) the number of GPUs used by the given process. So:
- for `DataParallel`, `local_rank==-1` and `n_gpu` is the number of gpus of the machine used
- for `DistributedDataParallel`, `local_rank` is an integer between 0 and the number of GPUs of the machine - 1, and `n_gpu=1` (each process uses one GPU).

is it possible you wanted to put an `or` here?

Reply:
Oh, I see - thank you for catching that. So I didn't solve the actual problem, but had a lack of hiding it under the carpet.

The problem is that the `distributed=...` is wrong here - it is coded to expect ddp and not dp. dp doesn't have world_size/etc and so it fails, so should it be called `dpp` instead?
```
                self.train_dataset.make_sortish_sampler(
                    self.args.per_device_train_batch_size, dpp=self.args.local_rank != -1)
```
or re-coded to handle dp too? I don't know the initial intention.

In either case, we are back to the request at https://github.com/huggingface/transformers/issues/8822 - let's have a simple flag so that the downstream know which mode is it under and not via checking ranks and n_gpus which is very confusing.

Reply:
This got resolved with using: `distributed=(self.args.local_rank != -1),`